odoo/addons/base/tests/test_expression.py

```diff
@@ -1623,25 +1631,29 @@ def test_name_search(self):
         with self.assertQueries(['''
             SELECT "res_partner"."id"
             FROM "res_partner"
-            WHERE ("res_partner"."company_id" IN (
+            WHERE "res_partner"."company_id" IN (
                 SELECT "res_company"."id"
                 FROM "res_company"
-                WHERE ("res_company"."name"::text LIKE %s)
-            ))
+                WHERE "res_company"."name"::text LIKE %s
+            )
             ORDER BY "res_partner"."complete_name"asc,"res_partner"."id"desc
         ''']):
             self.Partner.search([('company_id', 'like', self.company.name)])
 
         with self.assertQueries(['''
             SELECT "res_partner"."id"
             FROM "res_partner"
-            WHERE (("res_partner"."company_id" IN (
-                SELECT "res_company"."id"
-                FROM "res_company"
-                WHERE (("res_company"."name"::text not like %s) OR "res_company"."name" IS NULL))
-            ) OR "res_partner"."company_id" IS NULL)
+            WHERE (
+                "res_partner"."company_id" IS NULL
+                OR "res_partner"."company_id" IN (
+                    SELECT "res_company"."id"
+                    FROM "res_company"
+                    WHERE ("res_company"."name"::text NOT LIKE %s OR "res_company"."name" IS NULL)
+                )
+            )
             ORDER BY "res_partner"."complete_name"asc,"res_partner"."id"desc
         ''']):
+            # XXX expecting to generate an IN for NOT LIKE
```

Comment:
It's partially done here, but not generalized. The test fails and produces a NOT IN.

Reply:
@ryv-odoo In this case, I have a query looking like:
`company_id is null or company_id not in (select from res_company where name like %s)`

This is because it should be complementary to the like condition which can include some constraints. Changing the condition to a in-notlike would not be equivalent.